pkg/image/qemu.go

```diff
@@ -33,48 +33,95 @@ const (
 	maxCPUSecs         = 30      //value from OpenStack Nova
 )
 
-// ConvertQcow2ToRaw converts a local qcou2 image to raw format
+// ConvertQcow2ToRaw converts a local qcow2 image to raw format
 func ConvertQcow2ToRaw(src, dest string) error {
-	err := execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
+	err := validate(src, "qcow2")
+	if err != nil {
+		return errors.Wrap(err, "image validation failed")
+	}
+
+	_, err = execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
 	if err != nil {
 		return errors.Wrap(err, "could not convert local qcow2 image to raw")
 	}
+
 	return nil
 }
 
-// ConvertQcow2ToRawStream converts an http accessible qcow2 image to raf format without locally caching the qcow2 image
+// ConvertQcow2ToRawStream converts an http accessible qcow2 image to raw format without locally caching the qcow2 image
 func ConvertQcow2ToRawStream(url *url.URL, dest string) error {
+	err := validate(url.String(), "qcow2")
+	if err != nil {
+		return errors.Wrap(err, "image validation failed")
+	}
+
 	jsonArg := fmt.Sprintf("json: {\"file.driver\": \"%s\", \"file.url\": \"%s\", \"file.timeout\": %d}", url.Scheme, url, networkTimeoutSecs)
-	err := execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", jsonArg, dest)
+	_, err = execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", jsonArg, dest)
 	if err != nil {
 		return errors.Wrap(err, "could not stream/convert qcow2 image to raw")
 	}
+
+	return nil
+}
+
+func validate(image, format string) error {
+	type imageInfo struct {
+		Format      string `json:"format"`
+		BackingFile string `json:"backing-filename"`
+	}
+
+	output, err := execWithLimits("qemu-img", "info", "--output=json", image)
```

Comment:
I do not know how `qemu-img` does reads on _qcow2_ files. But, does this func cause the entire (or a lot of?) the _qcow2_ file to be read?

Reply:
Pretty sure it doesn't read the whole file.  Would be surprised if it reads more than the header.

Reply:
right, it's just reading the header.